src/api/venue.ts

```diff
@@ -34,3 +44,71 @@ export const setVenueLiveStatus = async ({
     })
     .finally(onFinish);
 };
+
+// TODO: return an array of all venueId's we find on the full way up to the sovereignVenue
+export const fetchSovereignVenueId = async (
+  venueId: string,
+  previouslyCheckedVenueIds: readonly string[] = []
+): Promise<string> => {
+  const venue = await fetchVenue(venueId);
+
+  if (!venue) throw new Error(`The '${venueId}' venue doesn't exist`);
+
+  if (!venue.parentId) return venue.id;
+
+  if (previouslyCheckedVenueIds.includes(venueId))
+    throw new Error(
+      `Circular reference detected. '${venueId}' has already been checked`
+    );
+
+  return fetchSovereignVenueId(venue.parentId, [
+    ...previouslyCheckedVenueIds,
+    venueId,
+  ]);
+};
+
+export const fetchVenue = async (
+  venueId: string
+): Promise<WithId<AnyVenue> | undefined> => {
+  const venueDoc = await getVenueRef(venueId).get();
+
+  // TODO: Use proper data validation + firestore model converters to set this type rather than just forcing it with 'as'
+  //  see .withConverter(soundConfigConverter) in fetchSoundConfigs in src/api/sounds.ts as an example
+  const venue: AnyVenue | undefined = venueDoc.data() as AnyVenue | undefined;
```

Comment:
> TODO: Use proper data validation + firestore model converters to set this type rather than just forcing it with 'as'
> see `.withConverter(soundConfigConverter)` in `fetchSoundConfigs` in `src/api/sounds.ts` as an example

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1368/commits/684ea850233c48e27270daf046fd029c74aa2f3a